pkg/cloud-init/cloud-init.go

```diff
@@ -311,7 +311,7 @@ func ApplyMetadata(vm *v1.VM) {
 			return
 		}
 		// TODO Put local-hostname in MetaData once we get pod DNS working with VMs
-		msg := fmt.Sprintf("instance-id: %s.%s\n", domain, namespace)
+		msg := fmt.Sprintf("{ \"instance-id\": \"%s.%s\" }\n", domain, namespace)
```

Comment:
I had to change this because cirros requires that the meta-data be in json format instead of yaml. Cirros reports parsing errors otherwise and won't process the user-data.

Once we expand upon this structure and add more info, we may want to consider a struct that is encoded into json rather than manually formatting the json string as I've done here. I didn't see a reason to add that additional logic for the single 'instance-id' field.